server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -301,6 +309,29 @@ void executeRequest(
             System::nanoTime
         );
         ActionListener<SearchRequest> rewriteListener = listener.delegateFailureAndWrap((delegate, rewritten) -> {
+            // add a listener in the chain that receives onResponse (successful responses) and logs a count of
+            // shard failures that are being returned to users, using the "partial-results" search logger
+            // and the failure details using the TransportSearchAction logger
+            ActionListener<SearchResponse> loggingListener = delegate.delegateFailure((upstream, searchResponse) -> {
+                List<ShardSearchFailure> failuresToLog = extractShardFailuresToLog(searchResponse);
+                if (failuresToLog.size() > 0) {
+                    Supplier<?> messageSupplier = () -> String.format(
+                        Locale.ROOT,
+                        "Search Exception has [%d] shard failures for request [%s]. See [%s] logger logs for details.",
+                        searchResponse.getShardFailures().length,
+                        original,
+                        this.getClass().getCanonicalName()
+                    );
+                    SEARCH_LOGGER.warn(messageSupplier);
```

Comment:
I wonder if this log line is useful. Perhaps this could be INFO. I mean all it tells is that there are shard failures worth investigating, and sends you to another logger to find them?